File: example/write.cc
#include "person.pb.h"
#include "../recordio.h"

#include <iostream>
#include <string>


int main() {
  // Write
  std::ofstream ofs("example.recordio", std::ios::binary);
  recordio::RecordWriter writer(&ofs);

  std::string names[] = {"Smith", "Johnson", "William", "Jones", "Brown"};
  int age = 20;
  for (const auto& name : names) {
    Person person;
    person.set_name(name);
    person.set_age(age++);
    writer.WriteProtocolMessage(person);
  }
  writer.Close();
  return 0;
}

File: example/write.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-

from person_pb2 import Person
import recordio

p = Person(name="John", age=100)

names = ["Smith", "Johnson", "William", "Jones", "Brown"]
ages = range(20, 20 + len(names))

with open("example.recordio", "wb") as f:
    for name, age in zip(names, ages):
        person = Person(name=name, age=age)
        writer = recordio.RecordWriter(f)
        writer.WriteProtocolMessage(person)
